### PR TITLE
Fix the issue about imcomplete input on powershell

### PR DESCRIPTION
**Pull request by wangqianzou**
Imcomplete input on powershell
POO: https://progress.opensuse.org/issues/109091

- Related ticket: https://progress.opensuse.org/issues/109091
- Needles: NA
- Verification run: https://openqa.suse.de/tests/8864107

### CODE REVIEW

File: tests/x11/remote_desktop/windows_network_setup.pm [hunk 1]
```diff
@@ -32,6 +32,7 @@ sub run {
     $self->run_in_powershell(cmd => 'Get-NetIPAddress -InterfaceAlias $a.name', tags => 'win-remote-desktop');
     $self->run_in_powershell(cmd => 'New-NetIPAddress -InterfaceAlias $a.name -IPAddress 10.0.2.18 -PrefixLength 24 -DefaultGateway 10.0.2.2 -Confirm:$false', tags => 'win-remote-desktop');
     approve_network_popup;
+    sleep 3;
```
Comment (rfan1) [inline tests/x11/remote_desktop/windows_network_setup.pm:35]: wait_still_screen 3;

Comment (wangqianzou) [reply tests/x11/remote_desktop/windows_network_setup.pm]: sleep 3 is passed.

Comment (wangqianzou) [reply tests/x11/remote_desktop/windows_network_setup.pm]: I try to use wait_still_screen

File: tests/x11/remote_desktop/windows_network_setup.pm [hunk 2]
```diff
@@ -32,6 +32,7 @@ sub run {
     $self->run_in_powershell(cmd => 'Get-NetIPAddress -InterfaceAlias $a.name', tags => 'win-remote-desktop');
     $self->run_in_powershell(cmd => 'New-NetIPAddress -InterfaceAlias $a.name -IPAddress 10.0.2.18 -PrefixLength 24 -DefaultGateway 10.0.2.2 -Confirm:$false', tags => 'win-remote-desktop');
     approve_network_popup;
+    wait_still_screen 3;
```
Comment (rfan1) [inline tests/x11/remote_desktop/windows_network_setup.pm:35]: LGTM， could you please add some comments here why you add 'sleep‘ here？

Comment (wangqianzou) [reply tests/x11/remote_desktop/windows_network_setup.pm:35]: > Got it.

Comment (wangqianzou) [reply tests/x11/remote_desktop/windows_network_setup.pm:37]: Add some comments and tags.

File: tests/x11/remote_desktop/windows_network_setup.pm [hunk 3]
```diff
@@ -32,6 +32,9 @@ sub run {
     $self->run_in_powershell(cmd => 'Get-NetIPAddress -InterfaceAlias $a.name', tags => 'win-remote-desktop');
     $self->run_in_powershell(cmd => 'New-NetIPAddress -InterfaceAlias $a.name -IPAddress 10.0.2.18 -PrefixLength 24 -DefaultGateway 10.0.2.2 -Confirm:$false', tags => 'win-remote-desktop');
     approve_network_popup;
+    # Imcomplete input on powershell because executing command is too fast
+    # Poo#109091
```
Comment (rfan1) [inline tests/x11/remote_desktop/windows_network_setup.pm:36]: Suggest change:
We may miss some characters due to type command too fast after network popup, so wait several seconds as a workaround,
See poo#109091

**Review by rfan1 — APPROVED** [review]
LGTM

**Comment by rfan1** [comment]
@GraceWang571 
Could you please help review this PR?

File: tests/x11/remote_desktop/windows_network_setup.pm (hunk 3 above)
Comment (wangqianzou) [reply tests/x11/remote_desktop/windows_network_setup.pm]: updated.

**Review by GraceWang571 — CHANGES_REQUESTED** [review]
I see many test cases invoke `run_in_powershell`, I'd like to say add `wait_still_screen` in `run_in_powershell` would be a better choice.
You know there might be similar failure when invoking `run_in_powershell` without wait_still_screen before/ in it.

**Comment by jouyingbin** [comment]
@wangqianzou there is typo in your subject and git commit message.

typo: imcomplete - > incomplete

**Comment by rfan1** [comment]
> I see many test cases invoke `run_in_powershell`, I'd like to say add `wait_still_screen` in `run_in_powershell` would be a better choice. You know there might be similar failure when invoking `run_in_powershell` without `wait_still_screen` before OR in it.

Actually, I talked with Roy last week, and I had the same suggestion, however, for this single case, the issue was not related to powershell script, since the issue seems happen after:

sub approve_network_popup {
    # if there is the networks popup that ask's if it's ok to be discoverable, approve it
    wait_still_screen stilltime => 5, timeout => 10;
    assert_screen([qw(networks-popup-be-discoverable no-network-discover-popup)], 60);
    if (match_has_tag 'networks-popup-be-discoverable') {
        assert_and_click 'network-discover-yes';
        wait_screen_change(sub { send_key 'ret' }, 10);
    }
}

**Comment by rfan1** [comment]
> > I see many test cases invoke `run_in_powershell`, I'd like to say add `wait_still_screen` in `run_in_powershell` would be a better choice. You know there might be similar failure when invoking `run_in_powershell` without `wait_still_screen` before OR in it.
> 
> Actually, I talked with Roy last week, and I had the same suggestion, however, for this single case, the issue was not related to powershell script, since the issue seems happen after:
> 
> sub approve_network_popup { # if there is the networks popup that ask's if it's ok to be discoverable, approve it wait_still_screen stilltime => 5, timeout => 10; assert_screen([qw(networks-popup-be-discoverable no-network-discover-popup)], 60); if (match_has_tag 'networks-popup-be-discoverable') { assert_and_click 'network-discover-yes'; wait_screen_change(sub { send_key 'ret' }, 10); } }

BTW， add wait_still_screen to `un_in_powershell` may consume more time :)

**Comment by GraceWang571** [comment]
> Actually, I talked with Roy last week, and I had the same suggestion, however, for this single case, the issue was not related to powershell script, since the issue seems happen after:
> 
> sub approve_network_popup { # if there is the networks popup that ask's if it's ok to be discoverable, approve it wait_still_screen stilltime => 5, timeout => 10; assert_screen([qw(networks-popup-be-discoverable no-network-discover-popup)], 60); if (match_has_tag 'networks-popup-be-discoverable') { assert_and_click 'network-discover-yes'; wait_screen_change(sub { send_key 'ret' }, 10); } }

I am afraid I can't agree with you. As you can see the `approve_network_popup;` was invoked twice in windows_network_setup.pm. And Roy only made changes for one of them and then the test case passed. 

As you might know, [poo#109091](https://progress.opensuse.org/issues/109091) is a performance issue and can't be reproduced 100%. Our goal is to make the test code robust to avoid the failures caused by the possible performance.

**Comment by rfan1** [comment]
@wangqianzou 
As offline discussion with @GraceWang571,

I would suggest you add the workaround into function `approve_network_popup`. since We use this function more than once!

And if you have seen any performance issue with `run_in_powershell`, you may add the similiar workaround as well to this function. [as far as I know, we rarely hit that issue before :)]

**Comment by rfan1** [comment]
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/15017

@wangqianzou You can close this PR once you back from vacation.

**Pull request closed without merging**